From 08d473f11c9d17b99a7062bb365f2bc7a26fd301 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 11 Jan 2021 18:22:45 +0200
Subject: [PATCH] [SW-20719] pll: introduce timer like api

As a preparation for HBM code addition (polling function specifically),
add timestamp_get_us() API function which returns the timestamp value in
(u)micro seconds.

Change-Id: Ifefc99d3ac166be57a40288346ddedb0075863f3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 591dd2b..b201341 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -58,6 +58,11 @@
 	return ((u64)up2 << 32) + low;
 }
 
+u64 timestamp_get_us(void)
+{
+	return udelay_timestamp_get() / cycles_per_usec;
+}
+
 void udelay(unsigned int usec)
 {
 	u64 start, current, end = cycles_per_usec * usec;
diff --git a/common/pll.h b/common/pll.h
index 371bf25..ca7961f 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -57,6 +57,7 @@
 u32 hl_pll_timestamp_base_get(void);
 u32 hl_pll_get_freq(u32 pll_base, enum hl_pll_div_out div);
 void udelay(unsigned int usec);
+u64 timestamp_get_us(void);
 
 static inline void mdelay(unsigned int msec) { udelay(1000 * msec); }
 
